tunnel/dns/server.go

```diff
@@ -150,13 +150,22 @@ func (r *resolver) testSystemResolver() error {
 	return nil
 }
 
-func (r *resolver) getAddress(name string) (net.IP, error) {
+func (r *resolver) getHostnameIp(name string) (net.IP, bool) {
+	r.namesMtx.Lock()
+	defer r.namesMtx.Unlock()
 	canonical := strings.ToLower(name)
-	a, ok := r.names[canonical]
+	ip, found := r.names[canonical]
+	return ip, found
```

Comment:
This will work, but I'm curious to know why a mutex over a self-contained implementation from cmap?

Reply:
cmap would probably be fine as well, but we already had the mutex there, just not consistently applied, so I went for the minimally intrusive fix